.github/workflows/update-dependency.yaml

```diff
@@ -0,0 +1,47 @@
+---
+name: 'Update Dependencies'
+on:
+  workflow_dispatch:
+  schedule:
+    # every Monday
+    - cron: '0 0 * * 1'
+env:
+  GOPROXY: https://proxy.golang.org
+permissions:
+  contents: read
+jobs:
+  update-deps:
+    if: ${{ github.repository == 'awslabs/amazon-eks-ami' }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - uses: actions/setup-go@4.0.0
+        with:
+          go-version: '1.21.5'
+      - uses: actions/checkout@v4
+      - name: Update Dependencies
+        id: update_deps
+        run: |
+          make update-deps
```

Comment:
we don't have this target yet right? yank from https://github.com/kubernetes/cloud-provider-aws/blob/master/hack/update-deps.sh?

Reply:
Added!